native/src/main/java/io/ballerina/stdlib/graphql/runtime/engine/meta/ServiceAnalyzer.java

```diff
@@ -0,0 +1,215 @@
+        Resource resource = new Resource(complexity);
+        this.resourceMap.put(coordinate, resource);
+        analyzeType(remoteMethod.getType().getReturnType());
+    }
+
+    private void analyzeType(Type type) {
+        Type impliedType = TypeUtils.getImpliedType(type);
+        int tag = impliedType.getTag();
+        if (tag == TypeTags.SERVICE_TAG) {
+            analyzeServiceType((ServiceType) impliedType);
+        }
+        if (tag == TypeTags.UNION_TAG) {
+            analyzeUnionType((UnionType) impliedType);
+        }
+        if (tag == TypeTags.ARRAY_TAG) {
+            analyzeType(((ArrayType) impliedType).getElementType());
+        }
+        if (tag == TypeTags.TABLE_TAG) {
+            analyzeType(((TableType) impliedType).getConstrainedType());
+        }
+        if (tag == TypeTags.RECORD_TYPE_TAG) {
+            analyzeRecordType((RecordType) impliedType);
+        }
+        if (tag == TypeTags.MAP_TAG) {
+            analyzeType(((MapType) impliedType).getConstrainedType());
+        }
+        if (tag == TypeTags.STREAM_TAG) {
+            analyzeType(((StreamType) impliedType).getConstrainedType());
+        }
+    }
+
+    private void analyzeServiceType(ServiceType serviceType) {
+        for (ResourceMethodType resourceMethod : serviceType.getResourceMethods()) {
+            String typeName = serviceType.getName();
+            analyzeResourceMethod(resourceMethod, typeName);
+        }
+    }
+
+    private void analyzeUnionType(UnionType unionType) {
+        for (Type memberType : unionType.getMemberTypes()) {
+            if (memberType.getTag() != TypeTags.NULL_TAG || memberType.getTag() != TypeTags.ERROR_TAG) {
+                analyzeType(memberType);
+            }
+        }
+    }
+
+    private void analyzeRecordType(RecordType recordType) {
+        String typeName = recordType.getName();
+        for (Field field : recordType.getFields().values()) {
+            String coordinate = typeName + "." + field.getFieldName();
+            Resource resource = new Resource(defaultQueryComplexity);
+            this.resourceMap.put(coordinate, resource);
+            analyzeType(field.getFieldType());
+        }
+    }
+
+    public Map<String, Resource> getResourceMap() {
+        return this.resourceMap;
+    }
+
+    private Long getComplexity(BMap<BString, Object> resourceConfig) {
+        if (resourceConfig == null) {
+            return this.defaultQueryComplexity;
+        }
+        if (resourceConfig.containsKey(complexityKey)) {
+            return (Long) resourceConfig.get(complexityKey);
+        }
+        return this.defaultQueryComplexity;
+    }
+
+    private static String getRootTypeName(ResourceMethodType resourceMethod) {
+        if (resourceMethod.getAccessor().equals(SUBSCRIBE_ACCESSOR)) {
+            return SUBSCRIPTION;
+        }
+        return QUERY;
+    }
+
+    private static String getSchemaCoordinate(String typeName, ResourceMethodType resourceMethodType) {
```

Comment:
Any reason to mark this method as `static` ? I think this can just be `private` instance method.
```suggestion
    private String getSchemaCoordinate(String typeName, ResourceMethodType resourceMethodType) {
```

Reply:
This method does not depend on the state of the instance. So I don't think this should be an instance method.